### PR TITLE
feat: add component meta data

### PR DESCRIPTION
**Pull request by winchesHe**
Closes # 

## 📝 Description

add components.json when publish

the content like the following picture

image

> Add a brief description

## ⛳️ Current behavior (updates)

> Please describe the current behavior that you are modifying

## 🚀 New behavior

> Please describe the behavior or changes this PR adds

## Summary by CodeRabbit

- **New Features**
	- Enhanced path resolution and console output with colored messages.
	- Implemented automatic generation of components metadata with improved error handling.
	- Defined constants for directory paths and URLs.
	- Updated main function to handle file restoration and metadata generation.

### CODE REVIEW

File: packages/core/react/src/scripts/postbuild.js [hunk 1]
```diff
@@ -1,21 +1,78 @@
...
+const chalk = require('chalk');
+
+const coreDir = path.resolve(__dirname, '../..'); // Core directory path
+const packagesDir = path.resolve(coreDir, '../..'); // Packages directory path
+const componentsDir = path.resolve(packagesDir, 'components'); // Components directory path
+const outputPath = path.resolve(coreDir, 'dist'); // Output directory path
+
+const rootDir = path.resolve(__dirname, '../../../../..'); // Root directory path
+const appsConfigDir = path.resolve(rootDir, 'apps/docs/config'); // Apps config directory path
+const appsRoutesJsonPath = path.resolve(appsConfigDir, 'routes.json'); // Apps routes file path
```
Comment (jrgarciadev) [inline packages/core/react/src/scripts/postbuild.js:14]: @winchesHe I think we could move this script to the root inside the `scripts` folder

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

File: packages/core/react/src/scripts/postbuild.js (hunk 1 above)
Comment (winchesHe) [reply packages/core/react/src/scripts/postbuild.js:14]: Got

File: .changeset/twenty-deers-wink.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@nextui-org/react": patch
+---
+
+Add components meta data when publish
```
Comment (jrgarciadev) [inline .changeset/twenty-deers-wink.md:5]: @winchesHe as the `scripts` folder is not included in the build we can remove this changeset

**Review by jrgarciadev — CHANGES_REQUESTED** [review]

File: .changeset/twenty-deers-wink.md (hunk 2 above)
Comment (winchesHe) [reply .changeset/twenty-deers-wink.md:5]: fixed

**Review by jrgarciadev — APPROVED** [review]
Thank you! @winchesHe NextUI CLI is coming 🔥

**Pull request merged**